src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKAnnotationPluginDescriptor.java

```diff
@@ -0,0 +1,432 @@
+        userArgs.annotationGroupsToUse.forEach(s -> {
+            if (!discoveredGroups.containsKey(s)) {
+                throw new CommandLineException("Unrecognized annotation group name: " + s);
+            }
+        });
+    }
+
+    /**
+     * Get the list of default plugins used for this instance of this descriptor. Used for help/doc generation.
+     * <p>
+     * NOTE: this method does not account for disabled default filters and just return ALL default instances.
+     * The refactored interface in Barclay changes it's contract to allows returning a list with only 'enabled' default
+     * instances. We'll change the implementation when we integrate the updated interface.
+     */
+    @Override
+    public List<Object> getDefaultInstances() {
+        return new ArrayList<>(toolDefaultAnnotations.values());
+    }
+
+    /**
+     * Pass back the list of Annotation instances that were actually seen on the command line in the annotations
+     * originating from groups first, followed by individual annotaitons in order. Its possible for this to return
+     * a filter that was originally included in the list of tool defaults only in the case where the user also
+     * specifies it on the command line.
+     * <p>
+     * NOTE: this method is somewhat misnamed in that it doesn't return ALL instances since it leaves out
+     * default filters (Except as noted above). The refactored interface in Barclay renames this method and
+     * changes it's contract. We'll change the implementation when we integrate the updated interface.
+     */
+    @Override
+    public List<Annotation> getAllInstances() {
+        final LinkedHashSet<Annotation> annotations = new LinkedHashSet<>();
+        userArgs.annotationGroupsToUse.forEach(s -> {
+            List<Annotation> as = discoveredGroups.get(s);
+            annotations.addAll(as);
+        });
+        userArgs.annotationsToUse.forEach(s -> {
+            Annotation rf = allDiscoveredAnnotations.get(s);
+            annotations.add(rf);
+        });
+        return new ArrayList<>(annotations);
+    }
+
+
+    /**
+     * Return the class representing the instance of the plugin specified by {@code pluginName}
+     *
+     * @param pluginName Name of the plugin requested
+     * @return Class object for the plugin instance requested
+     */
+    @Override
+    public Class<?> getClassForInstance(final String pluginName) {
+        return allDiscoveredAnnotations.get(pluginName).getClass();
+    }
+
+    /**
+     * Merge the default filters with the users's command line read filter requests, then initialize
+     * the resulting filters. Specifically, unless the user disables all tool default annotations it will
+     * first add all the tool enabled annotations which were not individually blocked by the user and then
+     * adds in annotations defined by the users specified groups, then individual annotations.
+     *
+     * @return Single merged read filter.
```

Comment:
It is not a single merged annotation, but the ones to use. Maybe to be consistent with the future PR in barclay (https://github.com/broadinstitute/barclay/pull/38), it can be named `getResolvedInstances`.

In addition, it may be useful to have a `getVariantAnnotatorEngine` method to create directly a `VariantAnnotatorEngine` from the parsed arguments.

Reply:
I agree with updating the comment, but for now I think for now we should try to maintain as much structural similarity as possible between the GATK descriptors, rather than making this one try to match the future (broadinstitute/barclay#38). Once we merge (broadinstitute/barclay#38) in then we can change the structure of both at the same time.